api/src/main/java/org/pmiops/workbench/api/ConceptSetsController.java

```diff
@@ -221,18 +126,15 @@ private void addConceptsToSet(DbConceptSet dbConceptSet, List<Long> addedIds) {
       String workspaceId,
       Long conceptSetId,
       UpdateConceptSetRequest request) {
-    DbConceptSet dbConceptSet =
-        getDbConceptSet(workspaceNamespace, workspaceId, conceptSetId, WorkspaceAccessLevel.WRITER);
-    validateAndUpdateDbConceptSetConcept(dbConceptSet, request);
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    dbConceptSet.setLastModifiedTime(now);
-    try {
-      dbConceptSet = conceptSetService.save(dbConceptSet);
-      return ResponseEntity.ok(toHydratedConceptSet(dbConceptSet));
-      // TODO: add recent resource entry for concept sets [RW-1129]
-    } catch (OptimisticLockException e) {
-      throw new ConflictException("Failed due to concurrent concept set modification");
-    }
+    // Fail fast if request isn't valid
+    validateUpdateConceptSetConcepts(request);
+
+    workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
+
+    return ResponseEntity.ok(
+        conceptSetService.toHydratedConcepts(
```

Comment:
Since both at line 120 and here toHydratedConcept is being called after updateConeptSet, how about calling   **toHydratedConcept** inside an conceptSetService.updateConceptSet? Or maybe since service is taking the UpdateConceptSetRequest should we make the service return 'ConceptSet' where ever possible like update/create?

Reply:
Good idea. Making toHydratedConcept a private method in ConceptSetService.